specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-03-01-preview/AlertsManagement.json

```diff
@@ -0,0 +1,2162 @@
+          },
+          {
+            "$ref": "#/parameters/targetResourceGroup"
+          },
+          {
+            "$ref": "#/parameters/targetResourceType"
+          },
+          {
+            "$ref": "#/parameters/monitorService"
+          },
+          {
+            "$ref": "#/parameters/monitorCondition"
+          },
+          {
+            "$ref": "#/parameters/severity"
+          },
+          {
+            "$ref": "#/parameters/alertState"
+          },
+          {
+            "$ref": "#/parameters/smartGroupIdFilter"
+          },
+          {
+            "$ref": "#/parameters/includePayload"
+          },
+          {
+            "$ref": "#/parameters/pageCount"
+          },
+          {
+            "$ref": "#/parameters/alertsSortBy"
+          },
+          {
+            "$ref": "#/parameters/sortOrder"
+          },
+          {
+            "$ref": "#/parameters/timeRange"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully listed alert objects.",
+            "schema": {
+              "$ref": "#/definitions/alertsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/errorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ListAlerts": {
+            "$ref": "./examples/Alerts_List.json"
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.AlertsManagement/alerts/{alertId}": {
+      "get": {
+        "operationId": "Alerts_GetById",
+        "summary": "Get a specific alert.",
+        "description": "Get information related to a specific alert",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/scope"
+          },
+          {
+            "$ref": "#/parameters/alertId"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Returns the alert with the specified ID.",
+            "schema": {
+              "$ref": "#/definitions/alert"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/errorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GetById": {
+            "$ref": "./examples/Alerts_GetById.json"
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.AlertsManagement/alerts/{alertId}/changestate": {
```

Comment:
>alerts/{alertId} [](start = 51, length = 16)

I have double checked. We can create subscription level resources using PUT.. for ex: we have subscription level deployments, role assignments etc.. 

Also, how are we going to create alerts in the first place ? 

Please add PUT for create and Patch for updating the state

Reply:
User can't create alerts. User creats alert rules using Microsoft.Insights API. Alert Rules create these alert instances.  We are consuming the alert intsances. Our API provides a way to fetch these alert instances. Alerts are proxy resources. While user creating the alert rules, they can specify the different resources. So there is no way to create a alert. User can only create alert rules.  There is no need for PUT API.

Our API is already GA. https://docs.microsoft.com/en-us/rest/api/monitor/alertsmanagement/alerts/changestate

In GA Version we are using POST for change state.
Why are you enforcing to use Patch for change state? As we discussed we don't want to change the behavior of the API for new version. 

We are just making alerts as extension resource for more granular access. Currently if user doesn't have access at subscription then they can't fetch alerts. Here the extension resource will be, the resource used while creating the alert rules. So if user has access to that particular resource still they can see alerts for that resource.

Reply:
Sounds good. makes sense to me. I did miss on the part of that user can't create alerts.